### PR TITLE
Add Action mix-in

### PR DESCRIPTION
**Pull request by nweathersbee**
This brings in the `MuchRails::Action` mix-in which defines
common behaviors for all view actions.

### CODE REVIEW

**Comment by nweathersbee** [comment]
@kellyredding Wasn't sure if we should keep a "default" error message for now. Maybe just sent the param name down and drop the default message. 🤷‍♂️

File: lib/much-rails/action.rb [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+      }
+    end
+
+    def memoize_boolean_decode(method_name)
+      -> {
+        instance_variable_name = "@#{method_name}"
+        if instance_variable_get(instance_variable_name).nil?
+          instance_variable_set(
+            instance_variable_name,
+            MuchRails::Boolean::FALSE_VALUES.exclude?(params[method_name])
```
Comment (kellyredding) [inline lib/much-rails/action.rb]: Does this work with `MuchRails::Boolean` (aka MuchBoolean)? I don't think this is the API to convert boolean values. I think it is `MuchRails::Boolean.convert(params[method_name])`. See https://github.com/redding/much-boolean.

File: test/unit/action_tests.rb [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+    let(:params) {
+      {
+        name: "NAME",
+        entered_on: Date.current,
+        updated_at: Time.current.utc,
+        active: "true"
+      }
+    }
+    should "return the expected Result" do
```
Comment (kellyredding) [inline test/unit/action_tests.rb:221]: empty line before `should` needed.

File: test/unit/action_tests.rb [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+  class InitTests < ReceiverTests
+    desc "when init"
+    subject { receiver_class.new(params: {}) }
+
+    should have_imeths :on_call, :valid_action?, :successful_action?
+  end
+
+  class ReceiverCallMethodTests < UnitTests
+    desc "receiver #call method"
+    subject { receiver_class.new(params: params) }
```
Comment (kellyredding) [inline test/unit/action_tests.rb]: I'd combine these two contexts - they are virtually identical in subjects:

```suggestion
  class InitTests < ReceiverTests
    desc "when init"
    subject { receiver_class.new(params: params) }

    should have_imeths :on_call, :valid_action?, :successful_action?
```

**Review by kellyredding — APPROVED** [review]
@nweathersbee one comment about using proper MuchBoolean API and a few minor test comments. Update the tests accordingly. Otherwise this looks pretty good.

**Comment by kellyredding** [comment]
@nweathersbee looks like I need to update MuchBoolean to get it up to our latest conventions also.

File: lib/much-rails/action.rb (hunk 1 above)
Comment (nweathersbee) [reply lib/much-rails/action.rb]: It works, but `MuchRails::Boolean.convert(params[method_name])` is a cleaner, less-coupled approach.

**Pull request merged**